### ISSUE TITLE
CalDAV calender confused when using exception in regular dates

### ISSUE THREAD

**Issue by thetwom** [SimpleMobileTools/Simple-Calendar#325]
I tried the following in my CalDAV under Thunderbird :
- Setting up a regular date (in my case every third week, starting December 4th 2017, no end)
- Adapt only one of the dates (e.g. shift the date of the 15th January 2018 one week later)

Synchronize this with the simple-calendar and CalDAV: 
-> The dates are shown incorrectly (in my case the 1th January 2018 disappeared, and the later dates are shown at incorrect weeks) 

In summary, when I setup a regular date in Thunderbird and change only one of the dates, simple-calendar becomes confused.

The google calendar would show the dates correctly. 

My setup:
- calDAV-calender via nextcloud (the online calender shows the dates correctly)
- simple-calendar 3.1.0
- lineageos 14.1
- DavDroid 1.9.10-ose

best regards

**Comment by tibbi** [comment]
Hey, some caldav related things have been fixed in the latest version 3.2.1, retest whenever it becomes available please. I plan doing some nextcloud testing, looks like it has more issues.

**Comment by thetwom** [comment]
That was quick! Thank you very much for your efforts!!

Ok, I just tested version 3.2.1, unfortunately it is still incorrect. 
Here is what happens exactly when I use the scenario from my first post:
- The first two dates (4th and 25th December) are correct
- 1st January suddenly appears (should not be there)
- 15th January disappers as it should 
- 22nd January appears according to the change, however, it appears twice
- The following weeks are now in a three week rythm but based on the 22nd January which is incorrect.

Hope this helps finding the problem.

**Comment by tibbi** [comment]
How have you modified a single instance at a recurring event? Im at the nextcloud web but I cannot do that.

**Comment by thetwom** [comment]
Hi, I modified the single event in Thunderbird which was also connected to the nextcloud account.

**Comment by tibbi** [comment]
aha ok I see thanks, thought it was directly on nextcloud

**Comment by tibbi** [comment]
it should be better in 4.0.x

**Comment by thetwom** [comment]
Thanks very much for your efforts. I tried the new version (4.0.0) but it didn't fix the problem.

**Comment by tibbi** [comment]
retest with 6.0 please, many related things have changed

**Comment by thetwom** [comment]
I just tried again with Version 6.0.1 from fdroid. Unfortunately, the issue remains.

**Comment by tibbi** [comment]
try with 6.0.2 please once it is released and becomes available, some related issues will be fixed

**Comment by thetwom** [comment]
Thank you for your efforts. I just tried version 6.1.2 from Fdroid and the issue seems resolved now. Thanks for the great work!
